src/python_testing/TC_IDM_4_2.py

```diff
@@ -0,0 +1,587 @@
+        self.print_step(
+            10, "CR1 sends a subscription request to subscribe to a specific global attribute from all clusters on all endpoints.")
+
+        # Omitting endpoint to indicate endpoint wildcard
+        cluster_rev_attr_path = [(cluster_rev_attr)]
+
+        # Subscribe to global attribute
+        sub_cr1_step10 = await CR1.ReadAttribute(
+            nodeid=self.dut_node_id,
+            attributes=cluster_rev_attr_path,
+            reportInterval=(3, 3),
+            keepSubscriptions=False
+        )
+
+        # Verify that the subscription is activated between CR1 and DUT
+        asserts.assert_true(sub_cr1_step10.subscriptionId, "Subscription not activated")
+
+        # Verify attribute data came back
+        self.verify_attribute_data(
+            sub=sub_cr1_step10,
+            cluster=Clusters.BasicInformation,
+            attribute=cluster_rev_attr
+        )
+
+        # Verify DUT sends back the attribute values for the global attribute
+        cluster_revision_attr_value = sub_cr1_step10.GetAttribute(cluster_rev_attr_typed_path)
+
+        # Verify ClusterRevision is of uint16 type
+        asserts.assert_true(self.is_uint16(cluster_revision_attr_value), "ClusterRevision is not of uint16 type.")
+
+        # Verify valid ClusterRevision value
+        asserts.assert_greater_equal(cluster_revision_attr_value, 0, "Invalid ClusterRevision value.")
+
+        sub_cr1_step10.Shutdown()
+
+        '''
+        ##########
+        Step 11
+        ##########
+        '''
+        self.print_step(11, "CR1 sends a subscription request to subscribe to a global attribute on an endpoint on all clusters.")
+
+        # Specifying single endpoint 0
+        cluster_rev_attr_path = [(0, cluster_rev_attr)]
+
+        # Subscribe to global attribute
+        sub_cr1_step11 = await CR1.ReadAttribute(
+            nodeid=self.dut_node_id,
+            attributes=cluster_rev_attr_path,
+            reportInterval=(3, 3),
+            keepSubscriptions=False
+        )
+
+        # Verify that the subscription is activated between CR1 and DUT
+        asserts.assert_true(sub_cr1_step11.subscriptionId, "Subscription not activated")
+
+        # Verify attribute data came back
+        self.verify_attribute_data(
+            sub=sub_cr1_step11,
+            cluster=Clusters.BasicInformation,
+            attribute=cluster_rev_attr
+        )
+
+        # Verify DUT sends back the attribute values for the global attribute
+        cluster_rev_attr_typed_path = self.get_typed_attribute_path(cluster_rev_attr)
+        cluster_revision_attr_value = sub_cr1_step11.GetAttribute(cluster_rev_attr_typed_path)
+
+        # Verify ClusterRevision is of uint16 type
+        asserts.assert_true(self.is_uint16(cluster_revision_attr_value), "ClusterRevision is not of uint16 type.")
+
+        # Verify valid ClusterRevision value
+        asserts.assert_greater_equal(cluster_revision_attr_value, 0, "Invalid ClusterRevision value.")
+
```

Comment:
Not part of the test case (yet), but it seems like we should also verify that the device DOESN'T send back data for other endpoints. Would you mind adding that check here and in the test plan?